src/pykka/_proxy.py

```diff
@@ -221,29 +220,110 @@ def __getattr__(self, name):
 
         if attr_info.callable:
             if attr_path not in self._callable_proxies:
-                self._callable_proxies[attr_path] = CallableProxy(
+                self._callable_proxies[attr_path] = CallableProxyFactory(
                     self.actor_ref, attr_path
                 )
             return self._callable_proxies[attr_path]
         elif attr_info.traversable:
             if attr_path not in self._actor_proxies:
-                self._actor_proxies[attr_path] = ActorProxy(self.actor_ref, attr_path)
+                self._actor_proxies[attr_path] = ActorProxyMessageBuilder(
+                    self.actor_ref, attr_path
+                )
             return self._actor_proxies[attr_path]
         else:
-            message = messages.ProxyGetAttr(attr_path=attr_path)
-            return self.actor_ref.ask(message, block=False)
+            return messages.ProxyGetAttr(attr_path=attr_path)
 
-    def __setattr__(self, name, value):
+    def setattr(self, name):
         """
         Set a field on the actor.
 
         Blocks until the field is set to check if any exceptions was raised.
         """
+        attr_path = self._attr_path + (name,)
+        return lambda value: messages.ProxySetAttr(attr_path=attr_path, value=value)
+
+
+class ActorProxyBase:
```

Comment:
Missing docstring about how and why this should be used. But I see this is also the case for `ActorProxy` - so it might not be fair to put this on you as a PR author. But I do think this could make sense for the project overall.